docs/user/config-file/index.rst

```diff
@@ -1,27 +1,120 @@
-Configuration file
-==================
+Using a configuration file
+==========================
 
-In addition to using the admin panel of your project to configure your project,
-you can use a configuration file in the root of your project.
-The configuration file should be named ``.readthedocs.yaml``.
+Content and structure of documentation undergo big and small changes.
+And eventually, the configuration of a documentation project also changes.
+This means you need to be able to track these changes over time,
+and keep them up to date.
 
-.. note::
+In this article,
+we cover the major concepts of using a configuration file:
 
-   Some other variants like ``readthedocs.yaml``, ``.readthedocs.yml``, etc
-   are deprecated.
+Versioning the configuration
+  A documentation project and its configuration file live together in a Git repository
+  and are versioned together.
+
+Configuration as code
+  Configuration uses the same workflow as your source code,
+  including being reviewed and tested in a Pull Request.
+
+Options that are not found in the configuration file
+  Not everything is suitable for version-specific configuration,
+  like the Git repository where the config file is read after cloning.
+
+.. seealso::
+
+   :doc:`/guides/setup/configuration-file`
+      Practical steps to add a configuration file to your documentation project.
+
+   :doc:`/config-file/v2`
+      Reference for configuration file settings.
+
+
+Why version your project's configuration?
+-----------------------------------------
+
+Consider the following aspects of a documentation project:
+
+Build environments change 📦️
+  You may depend on a number of packages but your method for installing them changes.
+  What is installed, how it's installed and what installs can change,
+  especially across multiple verisons.
+
+  You might change between Pip and Poetry.
+  You might also jump between Python 2 and 3 or Python 3.8 and Python 3.10.
+
+Documentation tools change ⚙️
+  Using Sphinx? Using MkDocs? Or some other tool?
+  All these tools have their own configuration files and special ways to invoke them.
+  In order to switch between how you are invoking the tool and setting up its environment,
+  you will need external build configuration.
+
+Comparing changes over time
```

Comment:
Maybe a little emoji? :heart: 

We could have a :mantelpiece_clock: ? :)

Reply:
The previous emoji was a :bomb: to signal the disruptive nature.. but that might have looked too scary :)